src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -4707,6 +4707,9 @@ static SimdIntrinsic packedsimd_methods [] = {
 	{SN_Multiply},
 	{SN_Negate},
 	{SN_ReplaceLane},
+	{SN_ShiftLeft},
+	{SN_ShiftRightArithmetic},
```

Comment:
Changing this to
{SN_ShiftLeft, OP_SHL},
etc. will make this code:
```
	if (op != 0)
		return emit_simd_ins_for_sig (cfg, klass, op, c0, arg0_type, fsig, args);
```
automatically emit it, so the switch case will no longer be neccessary.

Reply:
I agree with Zoltan.